common/hil_slurm_client.py

```diff
@@ -31,9 +28,9 @@ def check_hil_interface():
     hil_client = hil_init()
 
 
-def hil_reserve_nodes(nodelist):
+def hil_reserve_nodes(nodelist, dest_project):
     '''
-    Cause HIL nodes to move from the Slurm loaner project to the HIL free pool.
+    Cause HIL nodes to move from the Slurm loaner project to a new HIL project.
```

Comment:
I think it would be worth `asserting` that the project exists in HIL. `assert dest_project in [output of hil list_projects]`